include/xsimd/arch/generic/xsimd_generic_memory.hpp

```diff
@@ -120,6 +156,42 @@ namespace xsimd
             return detail::load_unaligned<A>(mem, cvt, generic {}, detail::conversion_type<A, T_in, T_out> {});
         }
 
+        // scatter
+        namespace detail
+        {
+            template <typename T, typename A, typename U, typename V, size_t I>
+            inline void scatter(batch<T, A> const& src, U* dst,
+                                batch<V, A> const& index,
+                                ::xsimd::detail::index_sequence<I>) noexcept
+            {
+                dst[index.get(I)] = static_cast<U>(src.get(I));
+            }
+
+            template <typename T, typename A, typename U, typename V, size_t I,
+                      size_t... Is,
+                      typename = typename std::enable_if<sizeof...(Is) != 0>::type>
+            inline void
+            scatter(batch<T, A> const& src, U* dst, batch<V, A> const& index,
+                    ::xsimd::detail::index_sequence<I, Is...>) noexcept
+            {
+                dst[index.get(I)] = static_cast<U>(src.get(I));
+                kernel::detail::scatter<T, A, U, V>(
+                    src, dst, index, ::xsimd::detail::index_sequence<Is...>());
+            }
+        } // namespace detail
+
+        template <typename A, typename T, typename U, typename V>
+        inline void scatter(batch<T, A> const& src, U* dst,
+                            batch<V, A> const& index,
+                            kernel::requires_arch<generic>) noexcept
+        {
+            static_assert(batch<T, A>::size <= batch<V, A>::size,
```

Comment:
Given that the `get` operator requires a store for each get, I'd rather perform a single aligned store to scratch memory, then perform the scatter sequentially. that being said, I'd also love *not* needing that generic implementation if we have a specialization for each instruction set.